### PR TITLE
test(js): Convert RuleBuilder / OwnerInput to RTL

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Converts both RuleBuilder and OwnerInput since they're kinda dependent on each other

### CODE REVIEW

File: static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx [hunk 1]
```diff
@@ -57,7 +61,7 @@ describe('RuleBuilder', function () {
       // Teams in project
       teams: [TEAM_1],
     });
-    act(() => ProjectsStore.loadInitialData([project]));
+    ProjectsStore.loadInitialData([project]);
```
Comment (priscilawebdev) [inline static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:64]: RIP `act` 👏

File: static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx [hunk 2]
```diff
@@ -66,43 +70,34 @@ describe('RuleBuilder', function () {
...
-    // This is because after selecting, react-select (async) reloads
-    await tick();
-    wrapper.update();
-    expect(wrapper.find(RuleBuilder)).toSnapshot();
+    expect(container).toSnapshot();
   });
 
-  it('renders with suggestions', async function () {
-    const wrapper = mountWithTheme(
+  it.only('renders with suggestions', async function () {
```
Comment (priscilawebdev) [inline static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:99]:
```suggestion
  it('renders with suggestions', async function () {
```

File: static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx [hunk 3]
```diff
@@ -113,36 +108,32 @@ describe('RuleBuilder', function () {
...
     // Open the menu so we can do some assertions.
-    openMenu(wrapper, {name: 'owners', control: true});
-    await tick();
-    wrapper.update();
+    const ownerInput = screen.getByRole('textbox', {name: 'Rule owner'});
+    selectEvent.openMenu(ownerInput);
 
-    // Should have all 4 users/teams listed
-    expect(wrapper.find('IdBadge')).toHaveLength(4);
+    await waitForElementToBeRemoved(() => screen.queryByText('Loading...'));
```
Comment (priscilawebdev) [inline static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:114]: jfyk.. sometimes I have issues with the ellipsis in my tests and check only for the loading. example: 

`screen.queryByText(/loading/)`

File: static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx [hunk 4]
```diff
@@ -34,14 +40,12 @@ describe('RuleBuilder', function () {
 
   const TEAM_1 = TestStubs.Team({
     id: '3',
-    name: 'COOL TEAM',
```
Comment (evanpurkhiser) [inline static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:37]: teams don't have `name`

Comment (priscilawebdev) [reply static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:37]: did typescript help you here? :)

File: static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx (hunk 3 above)
Comment (evanpurkhiser) [reply static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:114]: 👍 yeah sometimes we use …

File: static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx (hunk 4 above)
Comment (evanpurkhiser) [reply static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:37]: if it was converted to TS it would :-)

Comment (priscilawebdev) [reply static/app/views/settings/project/projectOwnership/ruleBuilder.spec.jsx:37]: oh yes it is still `.jsx` 😄

File: static/app/views/settings/project/projectOwnership/ruleBuilder.tsx [hunk 5]
```diff
@@ -169,29 +173,26 @@ class RuleBuilder extends Component<Props, State> {
...
+            value={owners}
+            onChange={this.handleChangeOwners}
+            disabled={disabled}
+          />
+          <Button
             priority="primary"
             disabled={!isValid}
             onClick={this.handleAddRule}
             icon={<IconAdd isCircled />}
-            size="sm"
```
Comment (priscilawebdev) [inline static/app/views/settings/project/projectOwnership/ruleBuilder.tsx:194]: is this the default?

**Comment by priscilawebdev** [comment]
since some css was changed, do you mind adding a before an after screenshots?

**Comment by priscilawebdev** [comment]
but it looks good

**Comment by evanpurkhiser** [comment]
It actually looks the same. I just got rid of all the pre-`gap` stuff

File: static/app/views/settings/project/projectOwnership/ruleBuilder.tsx (hunk 5 above)
Comment (evanpurkhiser) [reply static/app/views/settings/project/projectOwnership/ruleBuilder.tsx:194]: the default is `md` but we were adding padding to the button to make it size md, so I just.. let it be size md

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**